File: frontend/src/types/health.ts
export type HealthCheckStatus = 'ready' | 'degraded' | 'missing'
export type SystemHealthStatus = 'ok' | 'degraded'

export interface HealthCheck {
  key: string
  label: string
  status: HealthCheckStatus
  detail: string
  location?: string | null
}

export interface FeatureHealth {
  key: string
  label: string
  status: HealthCheckStatus
  detail: string
}

export interface SystemHealth {
  status: SystemHealthStatus
  checked_at: string
  output_folder: string
  checks: HealthCheck[]
  features: FeatureHealth[]
}

File: backend/app/models/health.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel

HealthCheckStatus = Literal["ready", "degraded", "missing"]
HealthStatus = Literal["ok", "degraded"]


class HealthCheck(BaseModel):
    key: str
    label: str
    status: HealthCheckStatus
    detail: str
    location: str | None = None


class FeatureHealth(BaseModel):
    key: str
    label: str
    status: HealthCheckStatus
    detail: str


class SystemHealth(BaseModel):
    status: HealthStatus
    checked_at: datetime
    output_folder: str
    checks: list[HealthCheck]
    features: list[FeatureHealth]
